### PR TITLE
Update Mathematica recipe

### PR DESCRIPTION
**Pull request by brisk022**
This pull request addresses the following issues.

- Licensing is currently missing from the recipe. Assuming that majority of the installations will be on clusters, information about the licence manager server would be required
- Replace url with url_for_version in case subsequent versions are added
- Create .spack dir during installation. This directory may not be created automatically causing installation failure when spack is trying to copy the log file there.
- Symlink wolframscript. Most people running mathematica in the cluster environment would be using wolframscript binary. However, the installer does not symlink it, so full path is required to run it.
- Generate spec.yaml. This file may not be created automatically causing package registration failure.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mathematica/package.py [hunk 1]
```diff
@@ -18,14 +18,38 @@ class Mathematica(Package):
...
             expand=False)
 
+    # Licensing
+    license_required = True
+    license_comment  = '#'
+    license_files    = ['Configuration/Licensing/mathpass']
+    license_url      = 'https://reference.wolfram.com/language/tutorial/RegistrationAndPasswords.html#857035062'
+
+    def url_for_version(self, version):
+        return "file://{0}/Mathematica_{1}_LINUX.sh".format(os.getcwd(), version)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mathematica/package.py]: This change is not necessary. Spack knows how to handle this URL:
```console
$ spack url parse file:///Users/Adam/spack/var/spack/repos/builtin/packages/Mathematica_12.0.0_LINUX.sh
==> Parsing URL: file:///Users/Adam/spack/var/spack/repos/builtin/packages/Mathematica_12.0.0_LINUX.sh

==> Matched version regex  0: r'^[a-zA-Z+._-]+[._-]v?(\\d[\\d._-]*)$'
==> Matched  name   regex  9: r'^([A-Za-z\\d+\\._-]+)$'

==> Detected:
    file:///Users/Adam/spack/var/spack/repos/builtin/packages/Mathematica_12.0.0_LINUX.sh
                                                              ----------- ~~~~~~
    name:    Mathematica
    version: 12.0.0

==> Substituting version 9.9.9b:
    file:///Users/Adam/spack/var/spack/repos/builtin/packages/Mathematica_9.9.9b_LINUX.sh
                                                              ----------- ~~~~~~
```

File: var/spack/repos/builtin/packages/mathematica/package.py [hunk 2]
```diff
@@ -18,14 +18,38 @@ class Mathematica(Package):
...
+
     def install(self, spec, prefix):
         sh = which('sh')
         sh(self.stage.archive_file, '--', '-auto', '-verbose',
            '-targetdir={0}'.format(prefix),
            '-execdir={0}'.format(prefix.bin),
            '-selinux=y')
+        # .spack directory in the install directory may not exist and needs to be created
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mathematica/package.py]: This shouldn't be the case. If it is, it's a bug that should be handled in Spack core. Can you open an issue about this with steps to reproduce it?

File: var/spack/repos/builtin/packages/mathematica/package.py [hunk 3]
```diff
@@ -18,14 +18,38 @@ class Mathematica(Package):
...
+        install_spack_dir = os.path.dirname(self.install_log_path)
+        os.makedirs(install_spack_dir, exist_ok=True)
+        # This is what most people would use on a cluster but the installer does not symlink it
+        ws_link_path = os.path.join(prefix.bin, 'wolframscript')
+        if not os.path.exists(ws_link_path):
+            ln = which('ln')
+            ws_path = os.path.join(prefix, 'Executables', 'wolframscript')
+            ln('-s', ws_path, ws_link_path)
+        # spec.yaml may not be generated automatically but required for successful registration of
+        # the package
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mathematica/package.py]: This shouldn't be the case. If it is, it's a bug that should be handled in Spack core. Can you open an issue about this with steps to reproduce it?

**Comment by brisk022** [comment]
I have created a bug report for the problem with missing .spack directory and spec.yaml #13718

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/mathematica/package.py:38: [E501] line too long (95 > 79 characters)
var/spack/repos/builtin/packages/mathematica/package.py:44: [W391] blank line at end of file
```

**Comment by adamjstewart** [comment]
Ping @brisk022

**Pull request merged**